brozzler/worker.py

```diff
@@ -499,11 +502,23 @@ def brozzle_site(self, browser, site):
                 # using brozzler-worker --proxy, nothing to do but try the
                 # same proxy again next time
                 logging.error("proxy error (self._proxy=%r)", self._proxy, exc_info=1)
-        except:
-            self.logger.error(
-                "unexpected exception site=%r page=%r", site, page, exc_info=True
-            )
+        except (brozzler.PageConnectionError, Exception) as e:
+            if isinstance(e, brozzler.PageConnectionError):
+                self.logger.error(
+                    "Page status code possibly indicates connection failure between host and warcprox: site=%r page=%r",
+                    site,
+                    page,
+                    exc_info=True,
+                )
+            else:
```

Comment:
Better to leave the default except block alone, and just have an except brozzler.PageConnectionError block above it